File: examfront/src/app/services/question.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {

  constructor(private _http: HttpClient) {}

  public getQuestionsOfSubject(qid: any){
    return this._http.get(`${baseUrl}/question/quiz/all/${qid}`); 
  }

  //add question

  public addQuestion(question:any)
  {
    return this._http.post(`${baseUrl}/question/`,question);
  }

  //delete question
  public deleteQuestion(quesId: any){
    return this._http.delete(`${baseUrl}/question/${quesId}`);
   }

   // add a single question

   public getQuestion(quesId: any){
    return this._http.get(`${baseUrl}/question/${quesId}`);
  }

  //update a question
  public updateQuestion(question:any){
    return this._http.put(`${baseUrl}/question/`,question);
  }


  //question required for the tes

  public getQuestionsOfSubjectForTest(qid: any){
    return this._http.get(`${baseUrl}/question/quiz/${qid}`); 
  }








}

File: examserver/src/main/java/com/exam/examserver/services/QuestionService.java
package com.exam.examserver.services;

import com.exam.examserver.entities.exam.Question;
import com.exam.examserver.entities.exam.Quiz;

import java.util.Set;

public interface QuestionService {


    public Question addQuestion(Question question);

    public Question updateQuestion(Question question);

    public Set<Question> getQuestions();

    public Question getQuestion(Long questionId);

    public Set<Question> getQuestionsOfQuiz(Quiz quiz);

    public void deleteQuestion(Long quesId);

    public Question get(Long questionsId);

}
